### ISSUE TITLE
Ignore caption when it is missing or blocked

### ISSUE THREAD

**Issue by hidde** [ableplayer/ableplayer#414]
Situation: a caption file is specified, but not available (returns a `404` or it does not load due to a CORS error or otherwise)

Expected:
- unavailability of caption is ignored

Actual result:
-  a JavaScript error logs into the console, AblePlayer buttons fail to load

I would like to request the broken caption to be ignored. Can try and submit a PR for it if that helps.

**Comment by paulbohman** [comment]
This is something that has affected us as well. Having some sort of error handler would be nice. We were attempting to load captions from a remote location using a full URL, and that fails. We can load a remote video, but we can't load remote captions, and that causes the whole media player to fail.

**Comment by hidde** [comment]
To add some more information: 

The error that's logged: 

```js
TypeError: can't access property "attr", this.$media is undefined
```

It seems to refer to the `attr()` call in line https://github.com/ableplayer/ableplayer/blob/master/scripts/buildplayer.js#L738.

**Comment by terrill** [comment]
In addition to the fix from @richard-smith-preservica one additional fix was needed: If only one track in a set of tracks fails to load, the failed track needs to be omitted from the CC popup menu. This is fixed in 92a47fd1a2960098469da64780d29af3a2ba5035
